pkg/dmi.go

```diff
@@ -0,0 +1,169 @@
+package pkg
+
+import (
+	"bufio"
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+type Entry = map[string][]string
+type DMIMap = map[string]Entry
+
+type DMIDecoder struct {
+	installed  bool
+	cmdOutput  string
+	decodedMap DMIMap
+}
+
+// NewCheckerService creates new instance from the parser
+func NewDMIDecoder() DMIDecoder {
+	dmidecode := exec.Command("sudo", "dmidecode")
+	res, _ := dmidecode.Output()
+
+	if len(res) > 0 {
+		return DMIDecoder{
+			installed:  true,
+			cmdOutput:  string(res),
+			decodedMap: DMIMap{},
+		}
+	} else {
+		fmt.Println("dmidecode is not installed in your device")
+		return DMIDecoder{}
+	}
+}
+
+// Decodes the string output of dmidecode
+func (dmi *DMIDecoder) Decode() {
+
+	if !dmi.installed {
+		return
+	}
+
+	scanner := bufio.NewScanner(strings.NewReader(dmi.cmdOutput))
+	key := ""
+	val := []string{}
+	parent := ""
+
+	// skip first 4 lines
+	for i := 0; i < 4; i++ {
+		scanner.Scan()
+	}
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if line == "" {
+			continue
+		}
+
+		if strings.HasPrefix(line, "Handle") {
+			key = ""
+			val = []string{}
+			parent = ""
+			continue
+		}
+
+		if string(line[0]) != "\t" && string(line[1]) != "\t" {
+
+			parent = line
+			dmi.decodedMap[parent] = Entry{}
+
+		} else if string(line[0]) == "\t" && string(line[1]) != "\t" {
+
+			line = strings.TrimLeft(line, "\t")
+			keyVal := strings.Split(line, ": ")
+
+			if len(keyVal) == 2 {
+
+				key = keyVal[0]
+				val = []string{keyVal[1]}
+
+			} else {
+
+				key = strings.ReplaceAll(keyVal[0], ":", "")
+				val = []string{}
+
+				if key == "System Power Controls" {
```

Comment:
let's make it generic to handle if the line after `Handle line` is starting with spaces, we ignore it and make a new section, I'm not sure if `System Power Controls` will be the only case